src/main.cpp

```diff
@@ -1328,13 +1326,13 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         return false;
     }
 
-    // perform some check related to sidechains state, e.g. creation of an existing scid, fw to
-    // a not existing one and so on
-    if (!scMgr.IsTxApplicableToState(tx) )
+    // If this tx creates a sc, no other tx must be doing the same in the mempool
+    if (!Sidechain::existsInMempool(pool, tx, state) )
```

Comment:
This call should be moved in the block below where the pool.cs is locked

Reply:
My medium term plan is to make Sidechain::existsInMempool a method of pool as the exists() one for ordinary transactions. This is why I chose to move the two methods close together.
I reckon a lock protection is needed; I'd rather fix the concurrencty issue calling LOCK(pool.cs) within Sidechain::existsInMempool. Any objection to it? thanks

Reply:
Agreed